[sca,fi] Update uJSON Bitstreams and Binaries

nasahlpa opened the pull request:
As the last update of bitstreams and binaries for the SCA and FI tests happened quite some time, this PR updates them to the versions specified in the commit message. Moreover, was there were some changes in the uJSON tests, the PR also updates some of the tests:
- Add device ID to HMAC test
- Configure iCache/dummy instrucitons
- Update num_segments

Finally, the documentation for building binaries and bitstreams have been updated.

nasahlpa commented:
@vogelpi FYI the binaries were created from my personal fork ([KMAC](https://github.com/nasahlpa/opentitan/tree/ot-sca-kmac), [AES](https://github.com/nasahlpa/opentitan/tree/ot-sca-aes)). Could you please have a look into these two branches and check if everything looks fine?

As already discussed offline, the traces for AES now look like this:
![image](https://github.com/user-attachments/assets/ec119e2c-c0fb-4743-8667-976cdeefe52e)

![image](https://github.com/user-attachments/assets/449a50e9-a281-4780-a7d1-54a4cc35270c)

But TVLA still looks fine:
![tvla_masks_off](https://github.com/user-attachments/assets/84bdf2c9-4e07-4f63-8fb7-25a47ae96a92)

andrea-caforio commented:
Looks sound.

Incidentally, what happened to our OT-SCA RFC, which I presented to the technical committee?

nasahlpa commented:
> Looks sound.
> 
> Incidentally, what happened to our OT-SCA RFC, which I presented to the technical committee?

Thanks for the review.

It got accepted and once we are sure that these branches for [KMAC](https://github.com/nasahlpa/opentitan/tree/ot-sca-kmac) and [AES](https://github.com/nasahlpa/opentitan/tree/ot-sca-aes) have all the changes we require, we will put them into the opentitan repo.

vogelpi commented on fault_injection/configs/pen.global_fi.ibex.char.flash_write.cw310.yaml:
```diff
@@ -42,4 +42,10 @@ test:
+  # When True, the dummy instructions are disabled. If False, use the default config
+  # (either on or off) set in the ROM.
+  dummy_instr_disable: False
+ 
```
There seems to be an issue here, not sure if it's a missing newline

nasahlpa replied:
Fixed

vogelpi commented on capture/configs/sha3_sca_cw310.yaml:
```diff
@@ -34,8 +34,8 @@ capture:
-  #which_test: sha3_fvsr_data
-  which_test: sha3_fvsr_data_batch
+  which_test: sha3_fvsr_data
+  #which_test: sha3_fvsr_data_batch
```
Is this change accidental or on purpose?

nasahlpa replied:
Thanks for spotting this, I've reverted the change.

vogelpi commented on capture/configs/aes_sca_cw310.yaml:
```diff
@@ -1,17 +1,15 @@
   # target_clk_mult is a hardcoded value in the bitstream. Do not change.
-  target_clk_mult: 0.24
-  target_freq: 24000000
+  target_clk_mult: 0.10
+  target_freq: 10000000
```
Yes, that's perfect!

vogelpi commented on capture/configs/aes_sca_cw310.yaml:
```diff
@@ -1,17 +1,15 @@
 target:
```
The commit message of this commit should probably be updated once we have pushed the changes into the upstream repo (in the form of the branches with tags as defined in the RFC).

nasahlpa replied:
Yes, my plan was to first have the new branches in lowRISC/opentitan and then update the commits here in ot-sca before merging it.

vogelpi commented on doc/building_fpga_bitstreams_binaries.md:
Please align the branch names with the agreed format as defined in the [RFC](https://docs.google.com/document/d/1JQ7X2a-25v2v05LvIPvPTEqMrON49yVh5vrMAzqaShg/edit?tab=t.0#heading=h.of98rfcbd97b)

vogelpi replied:
This makes sense to me. Would we still want to reduce the clock frequency? I can imagine for FI it makes a difference , too.

nasahlpa replied:
As most of the FI experiments anyway are performed on chips, I don't think that is really necessary. I would prefer leaving this ot-`sca_cw310` bitstream as close to upstream as possible to reduce the overhead for us.

vogelpi replied:
Okay, makes sense. And then the branch basically allows points to the commit on master where we've last generated this bistream.

vogelpi commented on doc/building_fpga_bitstreams_binaries.md:
Again align the branch name with the agreed format please.

vogelpi reviewed (approved):
This looks good. Thanks a lot for doing this Pascal!

Before merging this, can we please push the upstream branches and update binaries / commit messages?

nasahlpa commented:
> > @vogelpi FYI the binaries were created from my personal fork ([KMAC](https://github.com/nasahlpa/opentitan/tree/ot-sca-kmac), [AES](https://github.com/nasahlpa/opentitan/tree/ot-sca-aes)). Could you please have a look into these two branches and check if everything looks fine?
> 
> I did take a look and it makes sense to me. Two points worth discussing:
> 
> * The Commit "[[sca] Disable AES masking on CW310 to free FPGA resources](https://github.com/nasahlpa/opentitan/commit/d73eaa225ee07daa810408991d57350b87650f91)" is not really needed on the kmac / otbn branches as you've added a later commit to stub AES (great idea!!!). I would try to reduce the number of commits we need to maintain. However, splitting the addition of the stubbing parameter and enabling it makes sense to me.
> * I can't find the commit adding the stubbing parameter to SPI Host. Since I've added this and IIRC never upstreamed this, I suspect it's somewhere hidden in the git history. Can you please rebase your branches on master and make sure just the last couple of commits are really new on the ot-sca branches? I can also help you with that if you want, just let me know.
> 
> > As already discussed offline, the traces for AES now look like this: ![image](https://private-user-images.githubusercontent.com/5767400/421853210-ec119e2c-c0fb-4743-8667-976cdeefe52e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE4ODAxNzIsIm5iZiI6MTc0MTg3OTg3MiwicGF0aCI6Ii81NzY3NDAwLzQyMTg1MzIxMC1lYzExOWUyYy1jMGZiLTQ3NDMtODY2Ny05NzZjZGVlZmU1MmUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTNUMTUzMTEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2JlY2VjODI0MTk0YTQ3NmQ2NTMxNjNjMTlmZTczODEyM2ViY2E4ZTg4MTdlYjMxM2JjZWIxNDk4Yzk2OTcxMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.RvFRPKZ9kJYtpVKVob1mWcoXTYg9WOKXDbL7NjcUlgU)
> 
> I've seen this before and it's not concerning. The shape of the power trace keeps changing with different bitstreams and placement from time to time. What really matters is the alignment (plot below).
> 
> > ![image](https://private-user-images.githubusercontent.com/5767400/421853135-449a50e9-a281-4780-a7d1-54a4cc35270c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE4ODAxNzIsIm5iZiI6MTc0MTg3OTg3MiwicGF0aCI6Ii81NzY3NDAwLzQyMTg1MzEzNS00NDlhNTBlOS1hMjgxLTQ3ODAtYTdkMS01NGE0Y2MzNTI3MGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTNUMTUzMTEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWMxMTRmNDQxNTgzMzM2ZjljNDhkM2RhNzMwMmZkOGI0ZDc3NzljMjNiNDgwZThmZTlkODdiZDkxNTRiOTlmMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.C5paxXvDGs_CathSFIu3-i4kHkYQ3oG0YGl2CevPXho)
> 
> Fanatstic! The traces are now super clean.
> 
> > But TVLA still looks fine: ![tvla_masks_off](https://private-user-images.githubusercontent.com/5767400/421852805-84bdf2c9-4e07-4f63-8fb7-25a47ae96a92.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE4ODAxNzIsIm5iZiI6MTc0MTg3OTg3MiwicGF0aCI6Ii81NzY3NDAwLzQyMTg1MjgwNS04NGJkZjJjOS00ZTA3LTRmNjMtOGZiNy0yNWE0N2FlOTZhOTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTNUMTUzMTEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTY0MzdiNGIwZjc4Nzg5MDA0ZDQxNmExODYwZGFhNDljMTZiZGFhMDNlNmE5NDFjYjE2MTQ5MTc0NjViZjFkZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.15nrDAi74-7xI53t8aT4nQoAIoOr-j9xfVkn8JmjviI)
> 
> Great! If you have time, I would recommend doing a quick TVLA experiment where you sweep the number of traces used with 10 MHz and with 24 MHz to compare when the leakage kicks in. :-)

Thanks Pirmin.

I've renamed the branches to [ot-sca_cw310_kmac](https://github.com/nasahlpa/opentitan/tree/ot-sca_cw310_kmac) and [ot-sca_cw310_aes](https://github.com/nasahlpa/opentitan/tree/ot-sca_cw310_aes).

I've removed the AES masking commit from the ot-sca_cw310_kmac branch.

The SPI host stubbing is in the branches - I've rebased them.

regarding comparison - will do and post the results in the PR.

nasahlpa commented:
> 
> Great! If you have time, I would recommend doing a quick TVLA experiment where you sweep the number of traces used with 10 MHz and with 24 MHz to compare when the leakage kicks in. :-)

Here the plots for the bitstream & binary with 10MHz and reduced area:
![new_bitstream1](https://github.com/user-attachments/assets/0f6a111c-9e24-4aba-bade-4681a192df79)
![new_bitstream2](https://github.com/user-attachments/assets/9f34c3f2-2509-492c-bf40-a157f2eb5b5b)

And here the default bitstream and binary (upstream without any modifications):
![old_bitstream1](https://github.com/user-attachments/assets/afdd080e-fb91-4350-9b35-07a1439d1044)
![old_bitstream2](https://github.com/user-attachments/assets/7c12bed1-f707-48da-804b-1c8cceb146bd)

nasahlpa commented:
As I have updated the binaries & the commit message to point to the upstream branches, I am now merging this PR.
